src/common/util.js

```diff
@@ -182,44 +248,85 @@ function isNextLineEmptyAfterIndex(text, index) {
   }
   idx = skipTrailingComment(text, idx);
   idx = skipNewline(text, idx);
-  return hasNewline(text, idx);
+  return idx === false ? false : hasNewline(text, idx);
 }
 
+/**
+ * @template N
+ * @param {string} text
+ * @param {N} node
+ * @param {(node: N) => number} locEnd
+ * @returns {boolean}
+ */
 function isNextLineEmpty(text, node, locEnd) {
   return isNextLineEmptyAfterIndex(text, locEnd(node));
 }
 
+/**
+ * @param {string} text
+ * @param {number} idx
+ * @returns {number | false}
+ */
 function getNextNonSpaceNonCommentCharacterIndexWithStartIndex(text, idx) {
   let oldIdx = null;
-  while (idx !== oldIdx) {
-    oldIdx = idx;
-    idx = skipSpaces(text, idx);
-    idx = skipInlineComment(text, idx);
-    idx = skipTrailingComment(text, idx);
-    idx = skipNewline(text, idx);
-  }
-  return idx;
-}
-
+  /**
+   * @type {number | false}
+   */
+  let nextIdx = idx;
```

Comment:
However, if you need to override the inferred type like in this case, it's a valid reason to specify a type for a variable.

Reply:
Lets's use a more compact syntax:
```js
let nextIdx = /** @type {number | false} */ (idx);
```

Reply:
I would not favor adding extra parens unless absolutely necessary, which I think is not the case here. The following solution works for me (passes `yarn check-types`):

```js
  /** @type {number | false} */
  let nextIdx = idx;
```

Reply:
I would also favor adding something similar for the `let oldIdx` declaration on the line above, for the sake of someone reading the code for the first time, and some extra spacing. So I think the first few lines of the function could look something like this:

```js
function getNextNonSpaceNonCommentCharacterIndexWithStartIndex(text, idx) {
  /** @type {number | false} */
  let oldIdx = null;

  /** @type {number | false} */
  let nextIdx = idx;

  while (nextIdx !== oldIdx) {
    oldIdx = nextIdx;
```